### DISCUSSION TITLE
Unpacking GOG setup file on Linux

### DISCUSSION THREAD

**Discussion by phlash** [DescentDevelopers/Descent3#93] [General]
Hi folks, I have just purchased(!) the original D3 + Mercenary game on GOG to obtain the content files and thus wanted to unpack these files on my Linux (Debian 12) system so I can get going with the open source engine. Of course GOG only sell the Windows release, here's what worked for me:

 * Login to GOG.com
 * (optional) purchase the game :grin: 
 * Navigate to game details page
 * Do NOT hit that large blue button - use the 'Download Offline Backup Game Installers' link to the 'Descent 3 + Mercenary' file.
 * Check you download manager (there is no indication on the page itself - grumble grumble, javascript, grumble)
 * You should have `setup_descent_3_1.4_(16598).exe`, size ~0.8GB available...
 * (if required) Install the `innoextract` package
 * Create a temporary folder and `cd` to it
 * Unpack the setup: `innoextract <download folder>/setup_descent_3_1.4_(16598).exe`
 * You should see `app` and `tmp` folders
 * Move the `app` folder __contents__ to the game folder (CMake build seems to expect `~/Descent3` so I used: `mv app/* ~/Descent3`)
 * `cd` back up and nuke your temporary folder
 
[2024-05-06] __IMPORTANT: Ensure all HOG file names are _lower case_ or you will get object load failures!__:
 * `cd ~/Descent3`
 * ```for f in *.HOG; do t=`echo $f |tr '[A-Z]' '[a-z]'`; echo $f '->' $t; mv $f $t; done```

At this point I choose to sym-link the compiled binary and OS-specific hog, _your build path may vary_:
 * `ln -s ~/code/Descent3/build/Descent3/Descent3`
 * `ln -s ~/code/Descent3/build/Descent3/d3-linux.hog`

Enjoy!

**Comment by WTFreelance** [comment]
Try it with the --nosound option.
I.e.:
`cd ~/Descent3`
`./Descent3 --nosound`
